### PR TITLE
DNN-10289:Message/Email Tokens Don't Reflect SSL

### PR DESCRIPTION
**Pull request by Mohtshum**
- changes make in LocalizationProvider

### CODE REVIEW

File: DNN Platform/Library/Services/Localization/LocalizationProvider.cs [hunk 1]
```diff
@@ -97,7 +97,19 @@ public string GetString(string key, string resourceFileRoot, string language, Po
...
+            return string.IsNullOrEmpty(resourceValue) ? string.Empty : RemoveHttpUrlsIfSiteisSSLEnabled(resourceValue);
+        }
+
+        private string RemoveHttpUrlsIfSiteisSSLEnabled(string resourceValue)
+        {
+            var portalSettings = PortalController.Instance.GetCurrentPortalSettings();
+
+            if (portalSettings.SSLEnabled)
+            {
+                resourceValue = resourceValue.Replace(@"http:", @"https:");
```
Comment (Mohtshum) [inline DNN Platform/Library/Services/Localization/LocalizationProvider.cs]: I think resource entries should not be containing the external URLs. Either pointing to parent or child portals, which is taken care by GetCurrentPortalSettings

Comment (galatrash-at-dnn) [reply DNN Platform/Library/Services/Localization/LocalizationProvider.cs]: I agree. But your check is still missing one check for the other setting
``if (portalSettings.SSLEnabled || portalSettings.SSLEnforced)``

Comment (Mohtshum) [reply DNN Platform/Library/Services/Localization/LocalizationProvider.cs]: Ok thnx. Put that as well.

**Comment by kanm2015** [comment]
code review done

**Pull request merged**